### ISSUE TITLE
Enhancement: Allow to specify a custom text for comments to Pull requests

### ISSUE THREAD

**Issue by pacorreia** [infracost/infracost#1916]
Hi team,

Let me give you the following use case:
- Single project
- Terraform code using single module (root)
- Multiple environments specified by distinct terraform input var files

When I run Infracost under PR, I'm running it for all environments, meaning, specifying different input var files.

What happens when pushing comments back to devops is this:
![image](https://user-images.githubusercontent.com/54323159/181219894-6a046d63-d2ed-4997-8ed2-14e1e86330ee.png)

I know by order of execution which one is it, but would be really cool to have the chance to add a customized text or more detail to this comment to help distinguish between them

Regards and keep up with awesome work

**Comment by alikhajeh1** [comment]
@pacorreia great setup! Would [using `--project-name`](https://www.infracost.io/docs/features/cli_commands/#project-names) to set the text in the Project column help? Also, have you considered using a [config-file](https://www.infracost.io/docs/features/config_file/#usage) so you get 1 combined PR comment with both environments? The config-file also supports project name:

```
# you can also reference env vars in this file, e.g. workspace-${ENV_VARIABLE_HERE}
version: 0.1

projects:
  - path: code
    terraform_var_files:
      - stage.tfvars
    name: stage

  - path: code
    terraform_var_files:
      - prod.tfvars
      - us-east.tfvars
    name: prod
```

**Comment by pacorreia** [comment]
Will give it a try, thanks for pointing that out, will definitely try it and provide feedback

**Comment by alikhajeh1** [comment]
Awesome - looking forward to it :)

**Comment by pacorreia** [comment]
Just tested using a config file, although I've specified names for each, it picks my laptop names as project name for all in the output :| (tested locally first)

```yaml
version: 0.1

projects:
  - path: platform/src
    terraform_var_files:
      - ../vars/dev.json
    usage_file: platform/infracost-usage.yml
    name:hat-development

  - path: platform/src
    terraform_var_files:
      - ../vars/stag.json
    usage_file: platform/infracost-usage.yml
    name: hat-staging

  - path: platform/src
    terraform_var_files:
      - ../vars/prod.json
    usage_file: platform/infracost-usage.yml
    name: hat-production
```

And sample outcome:

> Project: LPT6331
> Module path: platform/src
> 
>  Name                                                                            Monthly Qty  Unit            Monthly Cost 
>                                                                                                                            
>  azurerm_application_insights.hat_appinsights["uksouth"]                                                                   
>  └─ Data ingested                                                                          3  GB                     $6.90 
>                                                                                                                            
>  azurerm_application_insights.hat_appinsights["ukwest"]                                                                    
>  └─ Data ingested                                                                          3  GB                     $6.90 
>                                                                                                                            
>  azurerm_cosmosdb_sql_container.postcode_deprivation_indexes_cosmosdb_container                                            
>  ├─ Provisioned throughput (provisioned, UK South)                                         4  RU/s x 100            $23.36 
>  ├─ Transactional storage (UK South)                                                       2  GB                     $0.50 
>  └─ Continuous backup (UK South)                                                           2  GB                     $0.48 
>                                                                                                                            
>  azurerm_cosmosdb_sql_database.hat_database                                                                                
>  ├─ Provisioned throughput (serverless)                                                0.002  1M RU                  $0.00 
>  ├─ Transactional storage (UK South)                                                       2  GB                     $0.50 
>  └─ Continuous backup (UK South)                                                           2  GB                     $0.48 
>                                                                                                                            
>  azurerm_log_analytics_workspace.default["uksouth"]                                                                        
>  └─ Log data ingestion                                                                     3  GB                     $6.90 
>                                                                                                                            
>  azurerm_log_analytics_workspace.default["ukwest"]                                                                         
>  └─ Log data ingestion                                                                     3  GB                     $6.90 
>                                                                                                                            
>  azurerm_storage_account.hat_test_reports["uksouth"]                                                                       
>  ├─ Capacity (first 50TB)                                                                  3  GB                     $0.14 
>  ├─ List and create container operations                                                 0.2  10k operations         $0.02 
>  ├─ Read operations                                                                     0.05  10k operations         $0.00 
>  └─ All other operations                                                                 0.1  10k operations         $0.00 
>                                                                                                                            
>  azurerm_storage_account.hat_test_reports["ukwest"]                                                                        
>  ├─ Capacity (first 50TB)                                                                  3  GB                     $0.14 
>  ├─ List and create container operations                                                 0.2  10k operations         $0.02 
>  ├─ Read operations                                                                     0.05  10k operations         $0.00 
>  └─ All other operations                                                                 0.1  10k operations         $0.00 
>                                                                                                                            
>  Project total                                                                                                      $53.24 
> 
> ──────────────────────────────────
> Project: LPT6331
> Module path: platform/src
> 
>  Name                                                                            Monthly Qty  Unit            Monthly Cost 
>                                                                                                                            
>  azurerm_application_insights.hat_appinsights["uksouth"]                                                                   
>  └─ Data ingested                                                                          3  GB                     $6.90 
>                                                                                                                            
>  azurerm_application_insights.hat_appinsights["ukwest"]                                                                    
>  └─ Data ingested                                                                          3  GB                     $6.90 
>                                                                                                                            
>  azurerm_cosmosdb_sql_container.postcode_deprivation_indexes_cosmosdb_container                                            
>  ├─ Provisioned throughput (provisioned, UK South)                                         4  RU/s x 100            $23.36 
>  ├─ Transactional storage (UK South)                                                       2  GB                     $0.50 
>  └─ Continuous backup (UK South)                                                           2  GB                     $0.48 
>                                                                                                                            
>  azurerm_cosmosdb_sql_database.hat_database                                                                                
>  ├─ Provisioned throughput (serverless)                                                0.002  1M RU                  $0.00 
>  ├─ Transactional storage (UK South)                                                       2  GB                     $0.50 
>  └─ Continuous backup (UK South)                                                           2  GB                     $0.48 
>                                                                                                                            
>  azurerm_log_analytics_workspace.default["uksouth"]                                                                        
>  └─ Log data ingestion                                                                     3  GB                     $6.90 
>                                                                                                                            
>  azurerm_log_analytics_workspace.default["ukwest"]                                                                         
>  └─ Log data ingestion                                                                     3  GB                     $6.90 
>                                                                                                                            
>  azurerm_storage_account.hat_test_reports["uksouth"]                                                                       
>  ├─ Capacity (first 50TB)                                                                  3  GB                     $0.14 
>  ├─ List and create container operations                                                 0.2  10k operations         $0.02 
>  ├─ Read operations                                                                     0.05  10k operations         $0.00 
>  └─ All other operations                                                                 0.1  10k operations         $0.00 
>                                                                                                                            
>  azurerm_storage_account.hat_test_reports["ukwest"]                                                                        
>  ├─ Capacity (first 50TB)                                                                  3  GB                     $0.14 
>  ├─ List and create container operations                                                 0.2  10k operations         $0.02 
>  ├─ Read operations                                                                     0.05  10k operations         $0.00 
>  └─ All other operations                                                                 0.1  10k operations         $0.00 
>                                                                                                                            
>  Project total                                                                                                      $53.24

**Comment by alikhajeh1** [comment]
@pacorreia what `infracost --version` are you using locally? (v0.10.8 is the [latest](https://www.infracost.io/docs/#1-install-infracost)).

**Comment by pacorreia** [comment]
yep that one precisely!, just double checked docs and it is supposed to work, I guess we just found out a bug by accident, maybe not many users are using this kinds of approach otherwise an issue would have been reported already.

This solution would be really really cool 👯

**Comment by alikhajeh1** [comment]
yeah that's odd! what's your operating system?
@aliscott can you take a look please?

**Comment by aliscott** [comment]
@pacorreia, hmmm this is definitely an odd one. Can you try the below and let me know.

1. What command are you using the to run Infracost?
2. On line 8 of your config file (`name:hat-development`), can you try adding a space after the `:`, i.e. `name: hat-development`
3. Can you run with `--format=json --out-file=infracost.json`. What does it show for the `name` and `metadata` keys for the projects in that JSON.
4. If you run `git ls-remote --get-url` in your code repo what do you see?

**Comment by pacorreia** [comment]
@aliscott 

I've that space after `:`, was a typo when removing sensitive information in my comment.

when running with JSON output there's  a sample from the file:

```json
  "projects": [
    {
      "name": "LPT6331",
      "metadata": {
        "path": "platform/src",
        "infracostCommand": "breakdown",
        "type": "terraform_dir",
        "terraformModulePath": "platform/src",
        "branch": "draft/test-pipeline",
        "commit": "9dc38305d9339ed8da801aec463a17511e5b176a",
        "commitAuthorName": "Paulo Correia",
        "commitAuthorEmail": "paulo.correia1@nhs.net",
        "commitTimestamp": "2022-07-27T11:38:31Z",
        "commitMessage": "Added infracost.yml",
        "vcsRepoUrl": "dev.azure.com/nhsuk/nhsuk.tools/_git/nhsuk.tools.heart-age",
        "vcsSubPath": "platform/src"
      },
```

About output of ls-remote:
https://xxx@dev.azure.com/xxx/xxx.tools/_git/xxx.xxxx.heart-age

**Comment by aliscott** [comment]
Thanks @pacorreia, can you also please let me know the full command you are using the to run Infracost?

**Comment by pacorreia** [comment]
@aliscott 
`infracost breakdown --config-file infracost.yml --format=json --out-file=./infracost.json`

**Comment by pacorreia** [comment]
Well, I'm not adding anything from you guys :)

no env vars, just checked.

Yes I'll grab proper time so we can crack on this one :) my mail is pcportugal@gmail.com, so you can expect some ping from that one, but will join slack for sure and reach you there

**Comment by aliscott** [comment]
@pacorreia and I paired on Zoom. The issue is that the `NAME` env variable was overwriting the config file's name. This was because we didn't have this ignored properly when loading the envs with envconfig. I've pushed a fix for this that we will get into the next release (https://github.com/infracost/infracost/pull/1919). I've also updated to enforce an `INFRACOST_` prefix for any of our envs going forward so this won't happen again accidentally.

**Comment by aliscott** [comment]
@pacorreia thanks for your help debugging this. I'll close this issue now and the fix will ship for this in v0.10.9 which should be sometime early next week.

**Comment by aliscott** [comment]
@pacorreia the `NAME` issue is now fixed in [v0.10.9](https://github.com/infracost/infracost/releases/tag/v0.10.9) release. Thanks for hopping on a Zoom call to help debug that 🙏.

**Comment by pacorreia** [comment]
cool, many thanks, great product and team
